### ISSUE TITLE
Batch size configurations for Transformer-XL benchmarks and convergence

### ISSUE THREAD

**Issue by carolynwang** [NVIDIA/DeepLearningExamples#973]
I see that the benchmarks and configurations provided in the Transformer-XL directory go up to a maximum global batch size of 512 (for the multi-node benchmark, 4/gpu * 16 gpus * 8 nodes). What was the reason for using a low batch size - is it because it cannot converge if it is larger?

**Comment by szmigacz** [comment]
Beyond batch size of 512 the scaling to achieve the target accuracy wasn't linear. For a single-node training of the large model, configs are using batch size = 128 and 100k steps, while preparing configs for multi-node I wanted to use the largest batch size that doesn't introduce additional "statistical inefficiency", we ended up with a batch size = 512 and 25k training steps (the batch size multiplied by 4 and the number of steps divided by 4). If I remember correctly batch size = 1024 required a bit more training steps (more than an ideal number of 12.5k steps) to achieve the target accuracy, I think it was converging after around 15k steps. Of course this is based on experiments with WT103 dataset which is relatively small, if you have a different, larger dataset then I'd expect that you could scale this codebase to a batch size much larger than 512 specified in config files for WT103.

**Comment by carolynwang** [comment]
Thank you!
